google/resource_compute_instance.go

```diff
@@ -135,10 +135,10 @@ func resourceComputeInstance() *schema.Resource {
 			},
 
 			"disk": &schema.Schema{
-				Type:       schema.TypeList,
-				Optional:   true,
-				ForceNew:   true,
-				Deprecated: "Use boot_disk, scratch_disk, and attached_disk instead",
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Removed:  "Use boot_disk, scratch_disk, and attached_disk instead",
```

Comment:
I'm okay with this but I just want to point it out. It might be good having it at the same time as the migration fn? That way we're saying "hey we're getting rid of this but hypothetically you don't need to do anything to replace it". I'm a bit nervous that we'll find a bug in the migration and people will be stuck though.

Reply:
My biggest concern is that if we don't remove it in the same release that we run the migration, we run the risk of people continuing to use it (despite it being deprecated) and then winding up with abandoned state (or having to run another migration). Basically, what if people upgrade, run the migration, and then continue to use the `disk` field, despite the deprecation?

Reply:
Yeah, I guess they're stuck in that case too.